From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Thu, 13 Jul 2023 18:08:05 +0200
Subject: [PATCH] Define Fiddle namespace

Adds a constant for the Fiddle namespace (for namespaced keys starting with `fiddle:`).
Its value is based on `NamespacedKey.FIDDLE`, which is defined in the "Define Fiddle namespace" patch in the `fiddle-api` module.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

diff --git a/src/main/java/net/minecraft/resources/ResourceLocation.java b/src/main/java/net/minecraft/resources/ResourceLocation.java
index fa9b84b604659a6b35262fbe5acbee0fdb20a56d..fa8f87b839ae42955ce8bd30b6fa55d7ac3036d3 100644
--- a/src/main/java/net/minecraft/resources/ResourceLocation.java
+++ b/src/main/java/net/minecraft/resources/ResourceLocation.java
@@ -26,6 +26,14 @@ public class ResourceLocation implements Comparable<ResourceLocation> {
     public static final char NAMESPACE_SEPARATOR = ':';
     public static final String DEFAULT_NAMESPACE = "minecraft";
     public static final String REALMS_NAMESPACE = "realms";
+    // Fiddle start - define Fiddle namespace
+    /**
+     * The namespace for Fiddle {@link ResourceLocation}s.
+     * This is for {@link ResourceLocation}s that are defined by and belong to Fiddle itself,
+     * not those of packs that are loaded by Fiddle (content in those packs uses its own namespaces).
+     */
+    public static final String FIDDLE_NAMESPACE = org.bukkit.NamespacedKey.FIDDLE;
+    // Fiddle end - define Fiddle namespace
     private final String namespace;
     private final String path;
 
